spec/swagger_spec.rb

```diff
@@ -11,8 +11,42 @@
 
   let(:swagger_routes) { Api::Docs.routes }
 
-  it "routes match" do
-    expect(rails_routes).to match_array(swagger_routes)
+  describe "Routing" do
+    include Rails.application.routes.url_helpers
+
+    it "routes match" do
+      expect(rails_routes).to match_array(swagger_routes)
+    end
+
+    it "allows customizable routes" do
+      with_env("PATH_PREFIX" => random_path) do
+        Rails.application.reload_routes!
+        expect(ENV["PATH_PREFIX"]).not_to be_nil
+        expect(api_v0x0_sources_url(:only_path => true)).to eq("/#{ENV["PATH_PREFIX"]}/v0.0/sources")
+      end
+      Rails.application.reload_routes!
+    end
+
+    def random_url_char
+      (("a".."z").to_a << "_").sample
+    end
+
+    def random_path_part
+      (1..20).to_a.sample.times.collect { random_url_char }.join
+    end
+
+    def random_path
+      (1..10).to_a.sample.times.collect { random_path_part }.join("/")
```

Comment:
Another fun way, just because...

```ruby
def random_path
  words = File.readlines("/usr/share/dict/words")
  rand(1..10).times.collect { rand(1..3).times.collect { words.sample.downcase.chomp }.join("_") }.join("/")
end
```

Reply:
good stuff...
```ruby
irb(main):012:0> puts random_path
bedrite_slabwood_heterothallism/slimsy_Rozalin_year-book_unemulative
=> nil
irb(main):013:0> puts random_path
heterogone_Strophomenacea_All-russia_Parcel/quasi-skillfully/Gallophobe_nondemocratic_Jacaltec_endocrinotherapy/perfective
```